### PR TITLE
GH-10: Add basic pre-commit-config with Apache RAT

### LINKED ISSUE #10: pre-commit: ShellCheck
See also: https://github.com/apache/arrow-go/blob/e112ad0de5ca66133256f2db71bda4dd2d1731f5/.pre-commit-config.yaml#L46-L49

### PR DESCRIPTION
**Pull request by lidavidm**
Fixes #10.

### CODE REVIEW

**Comment by lidavidm** [comment]
Ah, I should add the CI check too

**Review by kou — APPROVED** [review]
+1

**Comment by lidavidm** [comment]
Weird, it seems `--all-files` does not pass in other PRs, let me see...

**Comment by lidavidm** [comment]
I think the merge didn't account for pre-commit, should hopefully be fixed in #432

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+  - repo: local
+    hooks:
+      - id: rat
+        name: Release Audit Tool
+        language: system
+        entry: |
+          bash -c " \
+            git archive HEAD \
+              --prefix=apache-arrow-go/ \
```
Comment (kiszk) [inline .pre-commit-config.yaml:37]: I am not familiar with this tool. But is it better to use `java` instead of `go`?

Comment (kou) [reply .pre-commit-config.yaml:37]: Good catch!
We should use "java" here.

Comment (lidavidm) [reply .pre-commit-config.yaml:37]: Ah sorry, I copy-pasted and didn't check :sweat_smile:

**Pull request merged**